cluster/definition.yaml

```diff
@@ -0,0 +1,148 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xclusters.azure.platformref.crossplane.io
+  annotations:
+    upbound.io/ui-schema: |-
+      ---
+      configSections:
+      - title: Cluster Info
+        description: Information about this cluster
+        items:
+        - name: id
+          controlType: singleInput
+          type: string
+          path: ".spec.id"
+          title: Cluster ID
+          description: Cluster ID that other objects will use to refer to this cluster
+          default: platform-ref-azure-cluster
+          validation:
+          - required: true
+            customError: Cluster ID is required.
+        - name: writeSecretRef
+          controlType: singleInput
+          type: string
+          path: ".spec.writeConnectionSecretToRef.name"
+          title: Connection Secret Ref
+          description: name of the secret to write to this namespace
+          default: cluster-conn
+          validation:
+          - required: true
+      - title: Cluster Nodes
+        description: Enter information to size your cluster
+        items:
+        - name: clusterNodeCount
+          controlType: singleInput
+          type: integer
+          path: ".spec.parameters.nodes.count"
+          title: Node Count
+          description: Number of nodes in the cluster
+          default: 1
+          validation:
+          - minimum: 1
+          - maximum: 100
+          - required: true
+            customError: Node count is required.
+        - name: clusterNodeSize
+          controlType: singleSelect
+          path: ".spec.parameters.nodes.size"
+          title: Node Size
+          description: Desired node size (small, medium, or large).
+          default: small
+          enum:
+          - small
+          - medium
+          - large
+          validation:
+          - required: true
+            customError: Node size is required.
+      - title: Cluster Networking
+        description: Select a network fabric for your cluster
+        items:
+        - name: networkRef
+          controlType: singleInput
+          type: string
+          path: ".spec.parameters.networkRef.id"
```

Comment:
i believe the networkRef field isn't used on this type, so they should not be included in the UI metadata

Reply:
Makes sense. There are a number of things in the xrd and compositions that I plan to replace with the results of better harmony used on the multi platform repo pr.

Reply:
I think eliminating the use of networkRef and ClusterRef, replacing with single app-ref or maybe stack-ref, then using string transforms to name things like virtualnetwork, subnet, cluster, etc. would be more streamlined. Then the single app/stack value could be applied as a label to all resources and selectors would simply look for that More DRY'ish

Reply:
Captured a bit more of the broader discussion on network claims and refs in https://github.com/upbound/platform-ref-aws/issues/25#issuecomment-1006279269